tests/drivers/adc/adc_rescale/src/main.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 The ChromiumOS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/adc/adc_emul.h>
+#include <zephyr/drivers/adc/adc_voltage_divider.h>
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+
+/* Raw to millivolt conversion doesn't handle rounding */
+#define MV_OUTPUT_EPS 10
+
+#define ADC_TEST_NODE_0 DT_NODELABEL(sensor0)
+
+/**
+ * @brief Get ADC emulated device
+ *
+ * @return pointer to ADC device
+ */
+const struct device *get_adc_device(void)
+{
+	const struct device *const adc_dev = DEVICE_DT_GET(DT_NODELABEL(adc0));
+
+	zassert_true(device_is_ready(adc_dev), "ADC device is not ready");
+
+	return adc_dev;
+}
+
+ZTEST_USER(adc_rescale, test_get_adc_device)
+{
+	zassert_true(get_adc_device());
+}
+
+static int init_adc(const struct adc_dt_spec *spec, int input_mv)
+{
+	int ret;
+
+	zassert_true(device_is_ready(spec->dev), "ADC device is not ready");
+
+	ret = adc_channel_setup_dt(spec);
+	zassert_equal(ret, 0, "Setting up of the first channel failed with code %d", ret);
```

Comment:
why are we doing all this in a test? should the voltage divider API take care of ADC internally? I mean, given a "voltage-divider-dt-spec", I'd like to call something like `voltage_divider_get_voltage(&spec)`. How it's done internally, I don't care.

Reply:
I think (sorry been a while from the first review) I suggested to leave the actual adc details outside of this to avoid dealing with the sequencer configuration: some application probably wants to have the sequencer do all measurement together and just convert, some do not support the sequencer at all.